transformer_engine/common/util/cuda_runtime.cpp

```diff
@@ -145,6 +145,29 @@ const std::string &include_directory(bool required) {
   return path;
 }
 
+std::pair<int, int> stream_priority_range(int device_id) {
+  int ori_dev;
+  if (device_id >= 0) {
+    ori_dev = current_device();
+    NVTE_CHECK_CUDA(cudaSetDevice(device_id));
+  }
+  int min_priority, max_priority;
+  NVTE_CHECK_CUDA(cudaDeviceGetStreamPriorityRange(&min_priority, &max_priority));
+  if (device_id >= 0) NVTE_CHECK_CUDA(cudaSetDevice(ori_dev));
+  return std::make_pair(min_priority, max_priority);
+}
+
+bool supports_multicast(int device_id) {
+  int result;
+  CUdevice cudev;
+  if (device_id < 0) NVTE_CHECK_CUDA(cudaGetDevice(&device_id));
+  NVTE_CALL_CHECK_CUDA_DRIVER(cuDeviceGet, &cudev, device_id);
+  NVTE_CALL_CHECK_CUDA_DRIVER(cuDeviceGetAttribute, &result,
+                              CU_DEVICE_ATTRIBUTE_MULTICAST_SUPPORTED, cudev);
```

Comment:
This seems to be introduced in CUDA 12.1 and we support CUDA 12.0 - should be guarded.

Reply:
Also most probably for both of these functions we should not be calling those functions every single time, but rather cache the results.